source/declarations.tex

```diff
@@ -2868,159 +2868,132 @@
+
+\pnum
+\indextext{declaration!array}%
+In either case, the type is an \defn{array type}.
+The optional \grammarterm{attribute-specifier-seq}
+appertains to the array type.
+
+\pnum
+\tcode{T} is called the array \defn{element type};
+this type shall not be
+a placeholder type\iref{dcl.spec.auto},
+a reference type,
+\cv{}~\tcode{void}, or
+a function type.
+\begin{note}
+An array can be constructed from one of the fundamental types
+(except \tcode{void}),
+from a pointer,
+from a pointer to member,
+from a class,
+from an enumeration type,
+or from another array.
+\end{note}
+\begin{example}
+\begin{codeblock}
+float fa[17], *afp[17];
+\end{codeblock}
+declares an array of \tcode{float} numbers and
+an array of pointers to \tcode{float} numbers.
+\end{example}
+
+\pnum
 Any type of the form
-``\grammarterm{cv-qualifier-seq} array of
-\tcode{N}
-\tcode{T}''
+``\grammarterm{cv-qualifier-seq} array of \tcode{N} \tcode{T}''
 is adjusted to
-``array of
-\tcode{N}
-\grammarterm{cv-qualifier-seq}
-\tcode{T}'',
-and similarly for
-``array of unknown bound of
-\tcode{T}''.
-The optional \grammarterm{attribute-specifier-seq} appertains to the array.
+``array of \tcode{N} \grammarterm{cv-qualifier-seq} \tcode{T}'',
+and similarly for ``array of unknown bound of T''.
 \begin{example}
-
 \begin{codeblock}
 typedef int A[5], AA[2][3];
 typedef const A CA;             // type is ``array of 5 \tcode{const int}''
 typedef const AA CAA;           // type is ``array of 2 array of 3 \tcode{const int}''
 \end{codeblock}
 \end{example}
 \begin{note}
-An
-``array of
-\tcode{N}
-\grammarterm{cv-qualifier-seq}
-\tcode{T}''
+An ``array of \tcode{N} \grammarterm{cv-qualifier-seq} \tcode{T}''
 has cv-qualified type; see~\ref{basic.type.qualifier}.
 \end{note}
 
 \pnum
-An array can be constructed from one of the fundamental types
-(except
-\tcode{void}),
-from a pointer,
-from a pointer to member, from a class,
-from an enumeration type,
-or from another array.
-
-\pnum
-\indextext{declarator!multidimensional array}%
-When several ``array of'' specifications are adjacent, a
-multidimensional array
-type is created;
-only the first of
-the constant expressions that specify the bounds
-of the arrays may be omitted.
 In addition to declarations in which an incomplete object type is allowed,
 an array bound may be omitted in some cases in the declaration of a function
 parameter\iref{dcl.fct}.
 An array bound may also be omitted
-when the declarator is followed by an
-\grammarterm{initializer}\iref{dcl.init},
-when a declarator for a static data member is followed by a
-\grammarterm{brace-or-equal-initializer}\iref{class.mem},
-or in an explicit type conversion\iref{expr.type.conv}.
-In these cases, the bound is calculated from the number
+when an object (but not a non-static data member) of array type is initialized
+and the declarator is followed by an initializer
+(\ref{dcl.init}, \ref{class.mem}, \ref{expr.type.conv}).
```

Comment:
Hmm. So this continues to exclude [expr.new] because in `new (int[]) {1, 2, 3}` the declarator is not (immediately) followed by the initializer, and because `new int[] {1, 2, 3}` does not match the *new-type-id* grammar? (This is getting revised soon, but we need to preserve the existing behavior for now.)

Reply:
Let's make that explicit.